### PR TITLE
Add viewsetName to category when not a collection

### LINKED ISSUE #4176: `xml-editor`: Forms from different disciplines shown under the same header when creating new form based on existing one
Screenshot 2023-10-31 at 9 11 06 AM

When creating a new form based on an existing one, all forms are shown under the same discipline even thought they are distinct. This can be seen in the `nbm_mnb` database where the two preparation forms come from Ichthyology and Herpetology, but they are shown under the same header.

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4176

### CODE REVIEW

**Comment by CarolineDenis** [comment]
Before
Screenshot 2023-11-01 at 12 20 31 PM

After
Screenshot 2023-11-01 at 12 20 21 PM

File: specifyweb/frontend/js_src/lib/components/FormEditor/fetchAllViews.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ const augmentDatabaseView = (
       ? userInformation.availableCollections.find(
           ({ id }) => id === view.collectionId
         )?.collectionName
-      : undefined) ?? camelToHuman(view.viewsetLevel),
+      : undefined) ?? camelToHuman(`${view.viewsetLevel} ${view.viewsetName}`),
```
Comment (maxxxxxdlp) [inline specifyweb/frontend/js_src/lib/components/FormEditor/fetchAllViews.ts:81]: (optional) could we put some punctuation between level and name?

i.e
`${view.viewsetLevel} > ${view.viewsetName}`
or the nicer
`${view.viewsetName} (${view.viewsetLevel})`

**Review by maxxxxxdlp — APPROVED** [review]

**Review by grantfitzsimmons — CHANGES_REQUESTED** [review]
image

Instead of "Discipline Fish Views", can we make it accurate to the discipline the form definition comes from? 

In this case, it should say "Ichthyology Discipline" where `Ichthyology` comes from the `discipline.Name` field so that it is based on the actual name assigned by the user. 

Some people use "Botany" as the discipline template but manage pathogens or lichens in it instead. This would make the name more intuitive for users.

**Comment by CarolineDenis** [comment]
DECISION: 
Only display "Herpetology View" and loose Discipline.

**Review by grantfitzsimmons — APPROVED** [review]
💯 🚀

**Comment by maxxxxxdlp** [comment]
before merging this, @CarolineDenis can you test how dropping the viewsetLevel affect view sets from other levels?
if you are not sure, then proceed to only drop viewsetLevel when viewsetLevel is discipline

**Comment by CarolineDenis** [comment]
@maxxxxxdlp , @grantfitzsimmons , note that the discipline has been added as a title of the list in https://github.com/specify/specify7/pull/4190

**Review by pashiav — APPROVED** [review]
Looks good!
issue-4176

**Pull request merged**